mlir/lib/Quantum/Transforms/MergeRotationsPatterns.cpp

```diff
@@ -49,27 +52,111 @@ struct MergeRotationsRewritePattern : public mlir::OpRewritePattern<CustomOp> {
             return failure();
         }
 
-        TypeRange outQubitsTypes = op.getOutQubits().getTypes();
-        TypeRange outQubitsCtrlTypes = op.getOutCtrlQubits().getTypes();
-        ValueRange parentInQubits = parentOp.getInQubits();
-        ValueRange parentInCtrlQubits = parentOp.getInCtrlQubits();
-        ValueRange parentInCtrlValues = parentOp.getInCtrlValues();
-
-        auto parentParams = parentOp.getParams();
-        auto params = op.getParams();
-        SmallVector<mlir::Value> sumParams;
-        for (auto [param, parentParam] : llvm::zip(params, parentParams)) {
-            mlir::Value sumParam =
-                rewriter.create<arith::AddFOp>(loc, parentParam, param).getResult();
-            sumParams.push_back(sumParam);
-        };
-        auto mergeOp = rewriter.create<CustomOp>(loc, outQubitsTypes, outQubitsCtrlTypes, sumParams,
-                                                 parentInQubits, opGateName, nullptr,
-                                                 parentInCtrlQubits, parentInCtrlValues);
-
-        op.replaceAllUsesWith(mergeOp);
-
-        return success();
+        if (opGateName == "qml.Rot" || opGateName == "qml.CRot") {
+
+            if (op.getAdjoint()) {
+                LLVM_DEBUG(dbgs() << "Skipping adjointed operation:\n" << op << "\n");
+                return failure();
+            }
+
+            LLVM_DEBUG(dbgs() << "Applying scalar formula for combined rotation operation:\n" << op << "\n");
+            auto params = op.getParams();
+            auto parentParams = parentOp.getParams();
+
+            // Assuming params[0] = alpha1, params[1] = theta1, params[2] = beta1
+            // and parentParams[0] = alpha2, parentParams[1] = theta2, parentParams[2] = beta2
+
+            // Step 1: Calculate c1, c2, s1, s2
```

Comment:
Thanks for adding the new merge rotation pattern! The formula is very long, so we appreciate the good work :partying_face:

Reply:
Thank you so much for your help :) If there's anything specific you'd like me to refine or expand on, please let me know.